pkg/xds/server/server.go

```diff
@@ -93,21 +97,28 @@ func (values watches) Cancel() {
 	if values.secretCancel != nil {
 		values.secretCancel()
 	}
+	if values.runtimeCancel != nil {
+		values.runtimeCancel()
+	}
 }
 
 func createResponse(resp *cache.Response, typeURL string) (*v2.DiscoveryResponse, error) {
 	if resp == nil {
 		return nil, errors.New("missing response")
 	}
-	resources := make([]*types.Any, len(resp.Resources))
+	resources := make([]*any.Any, len(resp.Resources))
 	for i := 0; i < len(resp.Resources); i++ {
-		data, err := proto.Marshal(resp.Resources[i])
+		// Envoy relies on serialized protobuf bytes for detecting changes to the resources.
```

Comment:
On serialized bytes? Not on the version of the resource?

Reply:
copied "as is" from https://github.com/envoyproxy/go-control-plane/blob/34c8be4/pkg/server/server.go#L144